src/sinks/statsd/normalizer.rs

```diff
@@ -0,0 +1,454 @@
+use vector_core::event::{Metric, MetricValue};
+
+use crate::sinks::util::buffer::metrics::{MetricNormalize, MetricSet};
+
+#[derive(Default)]
+pub(crate) struct StatsdNormalizer;
+
+impl MetricNormalize for StatsdNormalizer {
+    fn normalize(&mut self, state: &mut MetricSet, metric: Metric) -> Option<Metric> {
+        // We primarily care about making sure that metrics are incremental, but for gauges, we can
+        // handle both incremental and absolute versions during encoding.
```

Comment:
As the current implementation stands, it doesn't do any of this, but I think we mostly get by because _most_ metrics sources will spit out incremental counters and so on.

Ironically, you can even see this in the `test_encode_absolute_counter` test, where it encodes an absolute counter, and has a code comment that the statsd parser will treat it as an incremental metric... so the current `statsd` sink behavior is demonstrably wrong, but again, most counters _are_ incremental so it just happens to generally work out fine and dandy.